src/test/validation_tests.cpp

```diff
@@ -55,15 +55,22 @@ BOOST_AUTO_TEST_CASE(block_subsidy_test)
 
 BOOST_AUTO_TEST_CASE(subsidy_limit_test)
 {
-    const auto chainParams = CreateChainParams(*m_node.args, ChainType::MAIN);
-    CAmount nSum = 0;
-    for (int nHeight = 0; nHeight < 14000000; nHeight += 1000) {
-        CAmount nSubsidy = GetBlockSubsidy(nHeight, chainParams->GetConsensus());
-        BOOST_CHECK(nSubsidy <= 50 * COIN);
```

Comment:
Why would you remove this check?

Reply:
> I do not like that the test no longer checks against max subsidy size and overflow/unreasonableness of the sum. What is your reasoning around that?

Thanks for the review! I aimed to view the code from the perspective of layman Bitcoin enthusiasts, trying to get an answer to their question, "How do we know the limit isn't greater than 21 million?".
I prioritized a high signal-to-noise ratio, omitting checks already covered by other tests. This test was about validating limits, not `GetBlockSubsidy` behavior, which was already tested elsewhere (e.g.,
[`TestBlockSubsidyHalvings`](https://github.com/bitcoin/bitcoin/blob/master/src/test/validation_tests.cpp#L33-L35)).

Here we're checking the sum in the end, so validating intermediary values was considered noise.
But if the reviewers insist, I can add it back.

Reply:
> I aimed to view the code from the perspective of layman Bitcoin enthusiasts

I don't think that many of them will look at our unit tests and if they do it's probably easier if you stick to writing a comment that is understandable. But I don't think they are the target audience of our unit tests.

> I prioritized a high signal-to-noise ratio

I don't think that's the right strategy for our unit tests in general. I don't think we should remove checks unless there is a clear reason to do so. Having a bit of duplication is fine as this is arguably the most important feature of bitcoin.

Reply:
I've added back the intermediary asserts.